src/SSHDebugPS/VSOperationWaiter.cs

```diff
@@ -22,15 +22,21 @@ internal static class VSOperationWaiter
         /// <exception cref="OperationCanceledException">Wait was canceled and 'throwOnCancel' is true</exception>
         public static bool Wait(string actionName, bool throwOnCancel, Action action)
         {
-            // TODO: Implement cancellationToken once remoteSystem.Connect(connectionInfo) handles it.
-            // Currently, the TWD will show up without a cancelation button but it will end in 30 seconds.
-            ThreadHelper.JoinableTaskFactory.Run(
-                actionName,
-                async (progress) =>
-                {
-                    await Task.Run(action);
-                }
-            );
+            try
+            {
+                //TODO: Once 'action' is cancelable, provide token to the action instead.
+                ThreadHelper.JoinableTaskFactory.Run(
+                    actionName,
+                    async (progress, cancellationToken) =>
```

Comment:
Note: you don't want to use `async` on your delegate. You aren't awaiting anything, so it isn't an async delegate.

Reply:
You'll `await` something once you implement my suggestion below. :)